File: 01-Assignment-02-02-2026/Using_C/program_19.c
/* Program 19: Find factorial of a number using recursion */
#include <stdio.h>

int factorial(int num){
    if (num == 0 || num == 1){
        return 1;
    }
    return num * factorial(num - 1);
}


int main() {
    int num;
    printf("Enter the Number : ");
    scanf("%d",&num);
    printf("Factorial of %d is %d",num,factorial(num));
    return 0;
}

File: 01-Assignment-02-02-2026/Using_Java/Program19.java
/*
Program 19
Java program to find the factorial of a number using recursion
*/

import java.util.Scanner;

public class Program19 {
    public static int factorial(int n){
        if (n == 1 || n == 0){
            return 1;
        }
        return n * factorial(n - 1);
    }

    public static void main(String[] args) {
        Scanner scanner =  new Scanner(System.in);
        System.out.print("Enter a Postive integer : ");
        int number = scanner.nextInt();

        System.out.println(number + "! => " + factorial(number));
        scanner.close();
    }

}
